### PR TITLE
ENG-5407 Simplify Python scripts with new SDK extensions

### PR DESCRIPTION
**Pull request by sebastian-quintero**
# Description

Uses the new Python SDK package extensions. For each Python app:

- Attempts to simplify the `main.py` as much as possible by using the new functionality.
- Updates the `requirements.txt` to only show what is needed (we don’t really need to show the numpy, pandas reqs if they are not necessary).

In addition, updates the test expectations.

There were some schema changes to the Python apps:
- Changes `solutions` for `solution`: this is the way it comes from the Nextmv Python SDK. Also we always have one solution, and this is a legacy name when we used to have multiple hop solutions.
- Added `options` key as it also comes from the Python SDK.
- Removed some information from some of the `statistics` and that info moved to options. For example: the provider comes as an option so it is recorded in the output’s `options` object, instead of being recorded in the statistics.

### CODE REVIEW

File: hello-world/main.py
Comment (merschformann) [inline hello-world/main.py:1]: Nice, you removed the `main`. I was asking to do that before, but we didn't want to break with the blog post. I suppose at this point that concern is gone anyway. :relaxed:

File: knapsack-highs/main.py [hunk 1]
```diff
@@ -2,134 +2,84 @@
...
 
-    start = time.time()
+    start_time = time.time()
+    nextmv.redirect_stdout()  # Solver chatter is logged to stderr.
 
     # Creates the solver.
     solver = highspy.Highs()
-    solver.silent()
-    solver.setOptionValue("time_limit", duration)
+    solver.silent()  # Solver output always logged to stdout, ignores redirect.
```
Comment (merschformann) [inline knapsack-highs/main.py]:
```suggestion
    solver.silent()  # Solver output ignores stdout redirect, silence it.
```

Just a suggestion for clarity. :shrug:

File: routing-ortools/main.py [hunk 2]
```diff
@@ -210,45 +196,35 @@ def capacity_callback(from_index):
...
-        "solutions": [{"vehicles": routes, "unplanned": []}],
-        "statistics": statistics,
-        "version": {"ortools": ortools_version},
-    }
+            ),
+        )
+
+    return nextmv.Output(
+        options=options,
+        solution={"vehicles": routes, "unplanned": []},
```
Comment (merschformann) [inline routing-ortools/main.py:225]: While I do agree with _not_ continuing to always return a list of solutions, this is breaking the expectations of the frontend route visualization (and immediate comparability to nextroute output).
Should we avoid the breaking change in the output? (for `routing-*` in particular) Or, should we adapt to it? Are these starter apps used by anyone in marketplace directly? :thinking:

**Review by merschformann — COMMENTED** [review]
Awesome work, thanks! :hugs: 

I only have some questions / concerns about the breaking output format changes (particularly on routing apps). I am happy to just approve it though, if this is in line with our decisions. :relaxed:

File: routing-ortools/main.py (hunk 2 above)
Comment (sebastian-quintero) [reply routing-ortools/main.py:225]: Per the discussion in standup: we are moving forward with the breaking change and simply releasing `v2` on the marketplace.

**Review by merschformann — APPROVED** [review]

**Pull request merged**